src/m365/spo/commands/sitedesign/sitedesign-get.ts

```diff
@@ -27,24 +31,57 @@ class SpoSiteDesignGetCommand extends SpoCommand {
     return 'Gets information about the specified site design';
   }
 
-  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
-    let spoUrl: string = '';
+  private getSiteDesignId(args: CommandArgs): Promise<string> {
+    if (args.options.id) {
+      return Promise.resolve(args.options.id);
+    }
+
+    const requestOptions: any = {
+      url: `${this.spoUrl}/_api/Microsoft.Sharepoint.Utilities.WebTemplateExtensions.SiteScriptUtility.GetSiteDesigns`,
+      headers: {
+        'X-RequestDigest': this.formDigestValue,
```

Comment:
We don't need request digest for this request, so let's remove it to avoid unnecessary calls.